### PR TITLE
Rework

### PR DESCRIPTION
**Pull request by chaosfreak93**
All Changes are tested on Motorola Moto G7 power (Android 10)

### CODE REVIEW

**Comment by chaosfreak93** [comment]
Don't merge

**Comment by chaosfreak93** [comment]
Don't merge

**Comment by chaosfreak93** [comment]
Can i add a German Translation?

**Comment by justdave** [comment]
I think it might be until the weekend before I get a chance to review this fully, but I really like what I see at first glance, thanks a bunch!!

**Comment by justdave** [comment]
First pass comments:

Not sure why Database.java and ServerDB.java are showing a delete+readd...  usually git will show that the file moved and preserve the history.  Did you use `git mv` on them?

You should add yourself to the about box.

**Comment by chaosfreak93** [comment]
> First pass comments:
> 
> Not sure why Database.java and ServerDB.java are showing a delete+readd... usually git will show that the file moved and preserve the history. Did you use `git mv` on them?
> 
> You should add yourself to the about box.

I've added me to the about box and i'm not using `git mv`

**Comment by chaosfreak93** [comment]
Don't merge

**Comment by justdave** [comment]
> i'm not using `git mv`

You should ;)  Not sure that can be fixed retroactively though.

**Comment by chaosfreak93** [comment]
> > i'm not using `git mv`
> 
> You should ;)  Not sure that can be fixed retroactively though.

Next time i will use it.

**Comment by justdave** [comment]
It would help me if you had a separate pull request for each thing you're working on.  Helps me review easier :-)  Feel free to make pull requests depend on each other so you don't have to start over with each one. Feel free to finish this one up with what's already in here (finish off the things you already started here).  A quick summary of what you've done would help me with what I'm looking at, as well.  Thanks again for helping out with this though!

**Comment by chaosfreak93** [comment]
I will explain all later but where you are setting the color of the text.

**Comment by chaosfreak93** [comment]
Btw sorry for my Bad english. I'm from Germany and try to not use Google translator. 🙃

**Comment by justdave** [comment]
> I will explain all later but where you are setting the color of the text.

https://github.com/chaosfreak93/MCStatus/blob/master/app/src/main/res/layout/about.xml#L55
^ for example.

I would probably export that to a style (one for light mode and one for dark mode?) and then use that same style for all of the dialogs, and pick the light-versus-dark at runtime when you load it.

**Comment by justdave** [comment]
There might be a system-defined style for those you could reference, too... I'd have to look it up. Though the system-defined style might be depending on newer Androids.

**Comment by chaosfreak93** [comment]
> There might be a system-defined style for those you could reference, too... I'd have to look it up. Though the system-defined style might be depending on newer Androids.

API-Level 21. I think

**Comment by justdave** [comment]
Looks like I can't force travis to re-merge this with master to test it without an additional commit to the PR :-(  In theory it ought to pass now if we could just make it test (restarting the build just rebuilds the same thing it already checked out instead of merging with master again)

**Comment by justdave** [comment]
For localization, instead of loading a different file based on the language, the docs suggest creating a new directory that has the language code on the end of the directory name, and then have the file of the same name in that directory. Android will then auto-pick and you won't need that if-then block.

If that doesn't work with "raw" resource types, then we can probably use https://developer.android.com/reference/android/content/res/Resources#getIdentifier(java.lang.String,%20java.lang.String,%20java.lang.String) and build a string to load it from, and fall back on the default if it's not there. That would make future languages a matter of dropping a file in instead of needing the if-then block for that as well.

**Comment by justdave** [comment]
Looks like we basically have 3 things in this PR right now:
* Rearrange the code, update for newer Android
* German Localization
* stubbed in settings dialog

Am I missing anything?  The last two look incomplete - I could merge the first right now if the other two were temporarily backed out.

Also, looks like the directory name thing does work with raw resources:

Screen Shot 2020-09-05 at 5 33 53 PM

**Comment by chaosfreak93** [comment]
It's midnight and I don't find a translation of "Privacy". The German Translation a probably done. Now we/I need to add the Settings Activity to change between Debug Message/Human Readable Messages and Light/Dark Mode. But for now, I will go to sleep. Good night

**Comment by chaosfreak93** [comment]
When the Settings Activity is done. You can merge and add the new Material

**Comment by justdave** [comment]
That's interesting. The licenses weren't accepted. I thought it was supposed to automatically accept the licenses unless you told it otherwise...

**Comment by justdave** [comment]
Meh.
```
Error: Ignoring unknown package filter 'android-30'
Warning: The package filter removed all packages. There is nothing to install.
```

So basically, Travis doesn't have Android-30 available to test against.  Which is odd.  build-tools-30.0.2 installed fine...

**Comment by justdave** [comment]
They must be using a cached mirror of the Google repositories, it can't pull a newer build-tools :|

**Comment by chaosfreak93** [comment]
Sorry, i have no more time for that. Fuck School😂

**Comment by justdave** [comment]
I figured out how to make Travis do API 30.  Will fix this up shortly.

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -2,7 +2,9 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="net.justdave.mcstatus">
 
-    <uses-permission android:name="android.permission.INTERNET"/>
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
Comment (justdave) [inline app/src/main/AndroidManifest.xml:7]: Lint complains that WRITE_EXTERNAL_STORAGE is no longer valid on Android 10+.  What are you using it for?

**Comment by justdave** [comment]
There's definitely three pieces to this, and it would be great if they could be separated so I can merge them to the main repo as three separate commits.

1. Project restructuring - this looks good to go now, but I can't merge it because of the other pieces attached to it. If you can get the other two pieces separated I can take this now. -> moved to #28 for merge
2. German localization - this also looks good to go, likewise can't pull it because the other is incomplete.
3. Settings dialog - this is the incomplete part.  The settings activity seems to work, but the setting changes don't appear to be saved yet.

I already moved 1 and 2 off to separate pull requests (see the references just above this comment).  The 3rd one barely had any work done in here, and it's easy to pick out.  Probably best starting from a new branch on that one if you want to continue it.

**Pull request closed without merging**